sky/packages/sky/lib/widgets/drawer.dart

```diff
@@ -55,70 +95,58 @@ class Drawer extends AnimatedComponent {
   DrawerStatusChangedCallback onStatusChanged;
   Navigator navigator;
 
-  AnimatedValue<Point> _position;
-  AnimatedColor _maskColor;
-  AnimationPerformance _performance;
+  SlideInIntention _intention;
+  ColorTransitionIntention _maskColorIntention;
+  AnimationPerformance get _performance => _intention.performance;
 
   void initState() {
-    _position = new AnimatedValue<Point>(_kClosedPosition, end: _kOpenPosition);
-    _maskColor = new AnimatedColor(colors.transparent, end: const Color(0x7F000000));
-    _performance = new AnimationPerformance()
-      ..duration = _kBaseSettleDuration
-      ..variable = new AnimatedList([_position, _maskColor])
-      ..addStatusListener(_onStatusChanged);
-    watch(_performance);
-    if (showing)
-      _show();
+    _intention = new SlideInIntention(
+        duration: _kBaseSettleDuration, start: _kClosedPosition, end: _kOpenPosition);
+    _maskColorIntention = new ColorTransitionIntention(
+        performance: _intention.performance, start: colors.transparent, end: const Color(0x7F000000));
```

Comment:
It's weird that one of these classes creates the performance and other one gloms on.  Won't they both try to `play` the performance?